ast-nodes/assignment.dart

```diff
@@ -12,10 +17,19 @@ class Assignment implements Statement {
 
   factory Assignment.parse(Iterable<Token> tokens) {
     var iter = tokens.iterator;
+
+    if (tokens.isEmpty) {
+      throw SyntaxError(iterator.current, "Expected an assignment");
+    }
+
     var tempLhs = consumeUntil(iter, RegExp("^:=\$"));
+
+    if (tempLhs.isEmpty) {
+      throw SyntaxError(iterator.current, "Expected an identifier");
+    }
+
     checkNext(iter, RegExp(':=\$'), 'Expected ":="');
```

Comment:
Previously we consumed until ":=" and not iterator current is ":=".
Check next will check next Token, which will be not ":=".
Maybe checkThis() or just skip this check.